### PR TITLE
Deprecate "Script-based Conditions" for removal

### LINKED ISSUE #1882: Deprecate "Script-based Conditions" for removal
Citing our [User Guide](https://junit.org/junit5/docs/current/user-guide/#writing-tests-conditional-execution-scripts)

> If you find yourself using the same script-based condition many times, consider writing a dedicated `ExecutionCondition` extension in order to implement the condition in a faster, type-safe, and more maintainable manner.

Additional reasons for removing "Script-based Conditions":
- #1481 -- Nashorn is being deprecated for removal
- https://github.com/sormuras/junit5-class-vs-module-path
- ...
- It was just an experiment.

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Closes #1882

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: buildSrc/src/main/kotlin/java-library-conventions.gradle.kts [hunk 1]
```diff
@@ -173,7 +173,7 @@ val compileModule by tasks.registering(JavaCompile::class) {
 tasks.compileJava {
 	// See: https://docs.oracle.com/en/java/javase/11/tools/javac.html
 	options.compilerArgs.addAll(listOf(
-			"-Xlint", // Enables all recommended warnings.
+			"-Xlint:all,-deprecation", // Enables all recommended warnings.
```
Comment (marcphilipp) [inline buildSrc/src/main/kotlin/java-library-conventions.gradle.kts]: Do we need to do this for all projects?

Comment (sormuras) [reply buildSrc/src/main/kotlin/java-library-conventions.gradle.kts]: No. It was a convenient place to redeem all temporarely introduced deprecation warnings at once.

Would we ship 5.5-M2 with the deprecated feature and remove it before 5.5 GA?

Comment (marcphilipp) [reply buildSrc/src/main/kotlin/java-library-conventions.gradle.kts]: No, the whole point of deprecating it first was to give users a chance to stop using it before we actually delete it. Thus, I think we should not remove it before 5.6/

Comment (sormuras) [reply buildSrc/src/main/kotlin/java-library-conventions.gradle.kts]: Alright. Then let's keep `-deprecation` active until `5.5 GA` is out and remove it immediately after starting with `5.6-SNAPSHOT`.

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledIf.java [hunk 2]
```diff
@@ -116,6 +116,7 @@
  * this annotation may be used in conjunction with other {@code @Enabled*} or
  * {@code @Disabled*} annotations in this package.
  *
+ * @deprecated for removal
```
Comment (marcphilipp) [inline junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledIf.java]: We should add a note about what to do instead here and for `@DisabledIf`.

Comment (sormuras) [reply junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledIf.java]: Good idea. Will add a sentence here and there.

File: buildSrc/src/main/kotlin/java-library-conventions.gradle.kts (hunk 1 above)
Comment (marcphilipp) [reply buildSrc/src/main/kotlin/java-library-conventions.gradle.kts]: Fair enough, please create an issue for us to do this for 5.6 M1.

Comment (sormuras) [reply buildSrc/src/main/kotlin/java-library-conventions.gradle.kts]: Done: https://github.com/junit-team/junit5/issues/1902

File: junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledIf.java (hunk 2 above)
Comment (sormuras) [reply junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledIf.java]: What about?

```
Script-based conditions APIs, and its supporting implementations
are deprecated with the intent to remove them in JUnit Jupiter 5.6.
Users should instead create and use a dedicated implementation of
{@link org.junit.jupiter.api.extension.ExecutionCondition ExecutionCondition}.
```

Comment (sbrannen) [reply junit-jupiter-api/src/main/java/org/junit/jupiter/api/condition/EnabledIf.java]: > Script-based condition APIs and their supporting implementations are deprecated with the intent to remove them in JUnit Jupiter 5.6. Users should instead rely on a combination of other built-in conditions or create and use a custom implementation of {@link org.junit.jupiter.api.extension.ExecutionCondition ExecutionCondition} to evaluate the same conditions.

**Pull request merged**